geopandas/io/sql.py

```diff
@@ -96,3 +96,222 @@ def load_geom_text(x):
                 crs = {"init": "epsg:{}".format(srid)}
 
     return GeoDataFrame(df, crs=crs, geometry=geom_col)
+
+
+def get_geometry_type(gdf):
+    """
+    Get basic geometry type of a GeoDataFrame,
+    and information if the gdf contains Geometry Collections."""
+    geom_types = list(gdf.geometry.geom_type.unique())
+    geom_collection = False
+
+    # Get the basic geometry type
+    basic_types = []
+    for gt in geom_types:
+        if "Multi" in gt:
+            geom_collection = True
+            basic_types.append(gt.replace("Multi", ""))
+        else:
+            basic_types.append(gt)
+    geom_types = list(set(basic_types))
+
+    # Check for mixed geometry types
+    assert len(geom_types) < 2, "GeoDataFrame contains mixed geometry types."
+    geom_type = geom_types[0]
+    return (geom_type, geom_collection)
+
+
+def get_srid_from_crs(gdf):
+    """
+    Get EPSG code from CRS if available. If not, return -1.
+    """
+    from pyproj import CRS
```

Comment:
Can you a TODO comment referring to #1101? This should be simplified once the other PR is merged.

Reply:
Done.

Reply:
@HTenkanen Can be simplified now, pyproj.CRS is merged in master now, so gdf.crs will always be pyproj.CRS or None.

Reply:
@martinfleis Excellent! I will update accordingly.

Reply:
@martinfleis Okay, simplified now the codes accordingly and things are now read from the pyproj.CRS of the GeoDataFrame directly. Also added Geoalchemy as a dependency for the Python 3.8 conda-forge yaml.